.github/workflows/ci.yaml

```diff
@@ -20,16 +20,21 @@ jobs:
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
-      - name: Mount bazel action cache
+      # Cache build and external artifacts so that the next ci build is incremental.
+      # Because caches cannot be updated, create a new cache based on the contents of
+      # hashFiles (modify this to capture sources in your repo that would cause bazel
+      # to rebuild artifacts, but it doesn't have to be perfect). On a cache miss, the
```

Comment:
Sources that are inputs to bazel. To minimize the amount of build work that would mean every source. But if it's onerous to declare every source in the hashFiles then it's not the end of the world to leave some out. You want enough sources so that the bulk of the bazel work will be cached. I'll try to reword.